---
int32 id
string message
bool success